dev/AppLifecycle/EncodedLaunchExecuteCommand.cpp

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "EncodedLaunchExecuteCommand.h"
+#include "ActivationRegistrationManager.h"
+
+using namespace winrt;
+using namespace winrt::Windows::Foundation;
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+    IFACEMETHODIMP EncodedLaunchExecuteCommand::Execute() noexcept try
+    {
+        THROW_HR_IF_NULL(E_INVALIDARG, m_selection);
+
+        com_ptr<IShellItem> item;
+        THROW_IF_FAILED(m_selection->GetItemAt(0, item.put()));
+
+        LPWSTR buffer = nullptr;
+        THROW_IF_FAILED(item->GetDisplayName(SIGDN_URL, &buffer));
+        std::wstring_view rawUri{ buffer };
+
+        Uri uri{ rawUri };
+        auto host = uri.Host();
+
+        auto separator = rawUri.find(L":");
+        if (separator == std::wstring::npos)
+        {
+            return E_INVALIDARG;
+        }
+
+        std::wstring encodedLaunchUri{ c_encodedLaunchSchemeName };
+        encodedLaunchUri += rawUri.substr(separator);
+
+        SHELLEXECUTEINFO executionInfo{};
+        executionInfo.cbSize = sizeof executionInfo;
+        executionInfo.fMask = SEE_MASK_CLASSNAME | SEE_MASK_FLAG_HINST_IS_SITE;
+
+        auto progId = ResolveUriTarget(encodedLaunchUri, host.c_str());
+        executionInfo.lpClass = progId.c_str();
+        executionInfo.lpFile = encodedLaunchUri.c_str();
+
+        com_ptr<::IUnknown> site;
+        THROW_IF_FAILED(GetSite(IID_PPV_ARGS(&site)));
+        executionInfo.hInstApp = reinterpret_cast<HINSTANCE>(site.get());
+
+        auto succeeded = ShellExecuteEx(&executionInfo);
+        if (!succeeded)
+        {
+            auto resultCode = reinterpret_cast<int64_t>(executionInfo.hInstApp);
```

Comment:
>int64_t [](start = 47, length = 7)

Is hInstApp 64bit on WIN32 platforms e.g. x86?

Shouldn't this be `reinterpret_cast<std::uintptr_t>(executionInfo.hInstApp);`

[https://stackoverflow.com/questions/153065/converting-a-pointer-into-an-integer#26586211](https://stackoverflow.com/questions/153065/converting-a-pointer-into-an-integer#26586211)

Reply:
This was a cut-n-paste from a totally different usage.  It's definitely wrong, but not for the reasons you think.  I'll update.  This member variable is special in that it's specifically a re-used compat value for old Win16 days.  So it's been documented to be treated a very specific way.  In our current usage this is never a pointer, but int64 is still the wrong size.